_posts/2020-08-24-python-calling-c++.md

```diff
@@ -0,0 +1,167 @@
+
+To understand why Python bindings are required let’s take a look at how Python and C++ store data and what types of issues this will cause. C or C++ stores data in the most compact form in memory possible. If you use an uint8_t, then the space required to store it would be approximately 8 bits if we don’t take structure padding into account. 
+
+In Python, on the other hand, everything is an object and the memory is heap allotted, integers in Python are big integers and their size may vary according to the value stored in them. This means that our Python bindings will need to convert a C/C++ integer to a Python integer for each integer passed across the boundary. 
+
+The process of transforming the memory representation of an object to a data format suitable for storage or transmission is called Marshalling and Python bindings are doing a process similar to Marshalling when they prepare data to move it from Python to C or vice versa. 
+
+##What is BoostPython library?  
+
+The Boost Python Library is a open source framework for interfacing Python and C++. It allows you to quickly and seamlessly expose C++ classes functions and objects to Python, and vice-versa, using no special tools, just your C++ compiler. It is designed to wrap C++ interfaces non-intrusively, so that you should not have to change the C++ code at all in order to wrap it, making Boost.Python ideal for exposing 3rd-party libraries to Python. The library's use of advanced metaprogramming techniques simplifies its syntax for users, so that wrapping code takes on the look of a kind of declarative interface definition language (IDL). It is designed to be minimally intrusive on your C++ design. In most cases, you should not have to alter your C++ classes in any way in order to use them with Boost.Python. The system should simply reflect your C++ classes and functions into Python. 
+
+The BoostPython Library binds C++ and Python in a mostly seamless fashion. It is just one member of the boost C++ library collection at [http://www.boost.org](http://www.boost.org). Boost.Python bindings are written in pure C++, using no tools other than your editor and your C++ compiler. 
+
+##Relationship to the Python C API 
+
+Python (more specific: CPython, the reference implementation of Python written in C) already provides an API for gluing together Python and C in the form of Python C API. Boost Python is a wrapper for the Python/C API. 
+
+Using the Python/C API, you must deal with passing pointers back and forth between Python and C and worry about pointers hanging out in one place when the object they point to has been thrown away. Boost Python takes care of much of this for you. In addition, Boost Python lets you write operations on Python objects in C++ in OOP style.
```

Comment:
do pointers really "hang out" or is this slang?

Reply:
It's a reference to dangling pointers. Dangling pointers are pointers pointing to a memory location that has been deleted